src/static/js/earn.js

```diff
@@ -1925,74 +1739,530 @@ async function main() {
       default:
         alert('Oops something went wrong. Try refreshing the page.');
     }
-  });
-
-  $(".approveBtn").click(function(){
-    let fn = $(this).attr("data-btn");
-    switch (fn) {
-      case 'approveSPGLSUSHI':
-        approveSPGLSUSHI();
-        break;
-      case 'approveSNOB':
-        approveSNOB();
-        break;
-      case 'approveSPGLPNG':
-        approveSPGLPNG();
-        break;
-      case 'approveSPGLETH':
-        approveSPGLETH();
-        break;
-      case 'approveSPGLUSDT':
-        approveSPGLUSDT();
-        break;
-      case 'approveSPGLLINK':
-        approveSPGLLINK();
-        break;
-      case 'approveS3D':
-        approveS3D();
-        break;
-      case 'approveS3F':
-        approveS3F();
-        break;
-      default:
-        alert('Oops something went wrong. Try refreshing the page.');
+  });
+
+  $(".claimBtn").unbind('click');
+  $(".approveBtn").click(function(){
+    let fn = $(this).attr("data-btn");
+    switch (fn) {
+      case 'approveSPGLSUSHI':
+        approveSPGLSUSHI();
+        break;
+      case 'approveSNOB':
+        approveSNOB();
+        break;
+      case 'approveSPGLPNG':
+        approveSPGLPNG();
+        break;
+      case 'approveSPGLETH':
+        approveSPGLETH();
+        break;
+      case 'approveSPGLUSDT':
+        approveSPGLUSDT();
+        break;
+      case 'approveSPGLLINK':
+        approveSPGLLINK();
+        break;
+      case 'approveS3D':
+        approveS3D();
+        break;
+      case 'approveS3F':
+        approveS3F();
+        break;
+      default:
+        alert('Oops something went wrong. Try refreshing the page.');
+    }
+  });
+
+  $(".stakeBtn").unbind('click');
+  $(".stakeBtn").click(function(){
+    let fn = $(this).attr("data-btn");
+    switch (fn) {
+      case 'stakeSPGLSUSHI':
+        stakeSPGLSUSHI();
+        break;
+      case 'stakeSNOB':
+        stakeSNOB();
+        break;
+      case 'stakeSPGLPNG':
+        stakeSPGLPNG();
+        break;
+      case 'stakeSPGLETH':
+        stakeSPGLETH();
+        break;
+      case 'stakeSPGLUSDT':
+        stakeSPGLUSDT();
+        break;
+      case 'stakeSPGLLINK':
+        stakeSPGLLINK();
+        break;
+      case 'stakeS3D':
+        stakeS3D();
+        break;
+      case 'stakeS3F':
+        stakeS3F();
+        break;
+      default:
+        alert('Oops something went wrong. Try refreshing the page.');
+    }
+  });
+
+  function stakeUnstake(amount, stake, st){
+    return `<div class="col-sm-12 col-md-3 align-items-center text-center snob-tvl pb-10 pb-md-0">
+    <p class="m-0 font-size-12"><ion-icon name="pie-chart-outline"></ion-icon> You have</p>
+    <p class="m-0 font-size-16 font-weight-regular">${amount} ${(st?st:'sPGL')} </p>
+    <p class="m-0 font-size-12">(Available to ${(stake? 'Stake': 'Unstake')}) </p>
+    </div>`
+  }
+  
+  function poolS3F(options){
```

Comment:
This is a huge function. Is it that long originally? Can we split it smaller? That's OK to leave it as is if it was already that long before

Reply:
It is long originally